src/internal.c

```diff
@@ -21166,7 +21166,8 @@ int ProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
                     word16 i = (word16)(ssl->buffers.inputBuffer.idx +
                                  ssl->curSize - ssl->specs.aead_mac_size);
 
-                    if (i > ssl->buffers.inputBuffer.length) {
+                    /* check i isn't too big and won't wrap around on --i */
+                    if (i > ssl->buffers.inputBuffer.length || i == 0) {
```

Comment:
If we need to test for integer overflow, then I suspect we need to test `ssl->buffers.inputBuffer.idx + ssl->curSize - ssl->specs.aead_mac_size` before casting it.